src/lines_to_params.rs

```diff
@@ -0,0 +1,165 @@
+        kwargs.set_item("source_actor", source_actor_obj).unwrap();
+        kwargs.set_item("target_actor", target_actor_obj).unwrap();
+        kwargs.set_item("ability", ability_event).unwrap();
+        kwargs.set_item("sequence_id", sequence).unwrap();
+
+        PyModule::import(py, "nari.types.event.ability").unwrap()
+            .getattr("Ability").unwrap().call((), Some(kwargs)).unwrap().to_object(py)
+    })
+}
+
+/// Params to action_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[int]")]
+pub(crate) fn action_effect_from_params(inp: Vec<&str>) -> ActionEffectParams {
+    let mut num = parser::u32_from_param(inp[0]);
+    let param0 = (&num >> 24) as u8;
+    let param1 = (&num >> 16) as u8;
+    let severity = (&num >> 8) as u8;
+    let effect_category = num as u8;
+    num = parser::u32_from_param(inp[1]);
+    let value = (&num >> 16) as u16;
+    let flags = (&num >> 8) as u8;
+    let multiplier = num as u8;
+    (param0, param1, severity, effect_category, value, flags, multiplier)
+}
+
+/// Params to status_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> (int, int, float, int")]
+pub(crate) fn status_effect_from_params(inp: Vec<&str>) -> StatusEffectParams {
+    let (param0, param1) = parser::u16x2_from_param(inp.get(0).unwrap());
+    (
+        param0,
+        param1,
+        parser::f32_from_param(inp.get(1).unwrap()),
+        parser::u32_from_param(inp.get(2).unwrap()),
+    )
+}
+
+/// Params to statuslist
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[any]")]
+pub(crate) fn statuslist_from_params(inp: Vec<&str>) -> StatuslistParams {
+    let mut col = inp;
+    let actor = col.drain(..2).collect::<Vec<&str>>();
+    let class = col.drain(..1).collect::<Vec<&str>>();
+    let resources = col.drain(..6).collect::<Vec<&str>>();
+    let position = col.drain(..4).collect::<Vec<&str>>();
+    let status_effects = col.drain(..col.len() - 1).collect::<Vec<&str>>();
+    (
+        parse_id_name_pair(actor),
+        class.first().unwrap(),
+        resources.iter().map(|x| parser::u32_from_param(x)).collect(),
+        position.iter().map(|x| parser::f32_from_param(x)).collect(),
+        status_effects
+            .chunks(3)
+            .map(|x| status_effect_from_params(x.to_vec()))
+            .collect(),
+    )
+}
+
+fn create_id_name_pair<'a>(py: Python<'a>, module: &'a str, class: &'a str, id_name_pair: Vec<&'a str>) -> &'a PyAny {
+    PyModule::import(py, module).unwrap()
+        .getattr(class).unwrap()
+        .call1(parse_id_name_pair(id_name_pair)).unwrap()
+}
+
+fn update_position(actor: &PyAny, position: Vec<f32>) {
+    actor.getattr("position").unwrap()
+        .call_method1("update", (
+            (&position[0]).to_owned(),
+            (&position[1]).to_owned(),
+            (&position[2]).to_owned(),
+            (&position[3]).to_owned())
+        ).unwrap();
+}
```

Comment:
The more I look at this the more it should just be a rust class just wrapped in python, instead of this half-ass not-exposed-but-messing-with-py thing. I don't see this even being used anywhere tho, and imo all the update position/resources stuff should be removed anyway.

Reply:
This is for fully constructing the Actor object for return to python without having an ugly export. Look 6 comments above.